app/models/course/settings/lesson_plan_items.rb

```diff
@@ -70,7 +66,26 @@ def lesson_plan_item_actable_names
     @components.map(&:class).map(&:lesson_plan_item_actable_names).flatten
   end
 
-  def disabled_tab_ids_for_lesson_plan
-    settings_interfaces_hash['course_assessments_component'].disabled_tab_ids_for_lesson_plan
+  # Gets the data needed for actable_hash from each component's settings_interface.
+  #
+  # For Assessments, return the tab IDs which are disabled.
+  #
+  # For Survey and Video where the setting is all or nothing, return nil if they're not supposed to
+  # be shown so the key isn't even in actable_hash. This is the same mechanism used to prevent items
+  # belonging to disabled components from showing in the lesson plan.
+  #
+  # @param [String] actable_name The name of the actable type.
+  # @return [Array<String> or nil]
+  def actable_hash_data(actable_name)
+    case actable_name
+    when Course::Assessment.name
+      [actable_name, settings_interfaces_hash['course_assessments_component'].disabled_tab_ids_for_lesson_plan]
```

Comment:
I didn't have time to review this in the previous PR but what you should do is not to store the logic of the hash here, but rather to call components that are 'lessson_plan_actable', and call their component/settings class to construct the hash. This way we keep the get/set logic of the leson_pan_actable settings in the same class so it is easier to refactor next time.